### ISSUE TITLE
Unexpected Exception: argument of type 'int' is not iterable

### ISSUE THREAD

**Issue by mearsten** [ansible/ansible#19520]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
`include_version` used in a loop statement

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = 
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
no specific configuration

##### OS / ENVIRONMENT
N/A

##### SUMMARY
If the name of the task is any integer >= 1 (without quotes) we get the exception mentioned in the header. If we write `- name: 0` everything works fine.

##### STEPS TO REPRODUCE
1) create file `roles/z1/tasks/main.yml`
```yaml
- debug: msg="included"
```

2) create file `roles/z2/tasks/main.yml`
```yaml
- name: 1
  include_role:
    name: z1
  with_items:
    - anything
```
3) create playbook `playbook.yml`
```yaml
- name: test
  hosts: localhost
  gather_facts: False
  roles:
    - z2
```

##### EXPECTED RESULTS
```
PLAY [test] ********************************************************************

TASK [z2 : include_role] *******************************************************

TASK [z1 : debug] **************************************************************
ok: [localhost] => {
    "msg": "included"
}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0   
```

##### ACTUAL RESULTS
```
PLAY [test] ********************************************************************
ERROR! Unexpected Exception: argument of type 'int' is not iterable
to see the full traceback, use -vvv
```

```
PLAYBOOK: playbook.yml *************************************************************
1 plays in playbook.yml

PLAY [test] ********************************************************************
ERROR! Unexpected Exception: argument of type 'int' is not iterable
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 103, in <module>
    exit_code = cli.run()
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/cli/playbook.py", line 159, in run
    results = pbex.run()
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 154, in run
    result = self._tqm.run(play=play)
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 282, in run
    play_return = strategy.run(iterator, play_context)
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/plugins/strategy/linear.py", line 178, in run
    host_tasks = self._get_next_task_lockstep(hosts_left, iterator)
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/plugins/strategy/linear.py", line 59, in _get_next_task_lockstep
    host_tasks[host.name] = iterator.get_next_task_for_host(host, peek=True)
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/executor/play_iterator.py", line 289, in get_next_task_for_host
    display.debug(" ^ task is: %s" % task)
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/playbook/task.py", line 143, in __repr__
    if self.get_name() == 'meta':
  File "/opt/ansible-2.2.0.0/local/lib/python2.7/site-packages/ansible/playbook/task.py", line 112, in get_name
    if self._role and self.name and ("%s : " % self._role._role_name) not in self.name:
TypeError: argument of type 'int' is not iterable
```

**Comment by alikins** [comment]
Not entirely sure if using ints as task/role names is expected to work or not. Either way, throwing an exception is a bug to fix.

**Comment by Akasurde** [comment]
@serge-name I am not able to reproduce this using latest Ansible version (2.7devel)
```
ansible-playbook play_z2.yml
PLAY [test] ********************************************************************************************************************

TASK [include_role : z1] *******************************************************************************************************

TASK [z1 : debug] **************************************************************************************************************
ok: [localhost] => {
    "msg": "included"
}

PLAY RECAP *********************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0

```

Could you please retry and let us know ? needs_info

**Comment by sivel** [comment]
This issue was recently resolved in 72ebd1bf4639ef021e733b1f70cd7e113add0c37 and will be included in the upcoming 2.6 release.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by Akasurde** [comment]
@sivel Thanks for the information.
